### PR TITLE
Support http-message 2.0 as well as 1.0.

### PR DESCRIPTION
**Pull request by NigelCunningham**
The 2.0 release only differs in adding return type hints and removing the strict types declarations. Either version should be usable with mpdf.

This update is also needed for projects that use pdf, where other dependencies already require http-message 2. We are seeking this need in the Drupal printable / pdf_api use of mpdf; the lack of http-message 2 support is preventing an update to Drupal 10.1 (https://www.drupal.org/project/printable/issues/3374990).

### CODE REVIEW

**Comment by finwe** [comment]
See [this comment](https://github.com/mpdf/mpdf/issues/1878#issuecomment-1569636010), make this a draft and propose only when certain composer ci on all supported PHP versions passes.

**Comment by NigelCunningham** [comment]
Yeah, that's what I started on - I initially thought the solution was going to be simpler since this one isn't overriding constructors; apologies for the premature PR. On the plus side, it hopefully provides visibility that something is happening and an opportunity for testers to jump in.

May I ask what might seem like a silly question from your perspective: why are you supporting such old versions of PHP?

**Comment by finwe** [comment]
There are people still using them.

**Comment by NigelCunningham** [comment]
Ok, and I can see there's no point going further with that question :)

**Comment by finwe** [comment]
Maybe not even with this PR, over the last months, I've become much closer to releasing the next major version with older PHP versions dropped.

**Comment by NigelCunningham** [comment]
Should I bother continuing then? I have plenty of other things I could spend the time on.

**Comment by finwe** [comment]
Depends on if and how fast you need this, I'd give the drop a few months (which is quite fast given how the library is evolving)

**Comment by NigelCunningham** [comment]
Ok. We have people wanting to use our Drupal module with D10 today, so I'll keep going.

**Comment by finwe** [comment]
I'll create the repository for the module and give you access, so you can work under mpdf namespace directly.

**Comment by NigelCunningham** [comment]
Oh, that would be great; thanks.

**Comment by finwe** [comment]
@NigelCunningham Can you please mark this as a draft until there is actually anything to review? I'm getting notified for every commit.

**Comment by NigelCunningham** [comment]
Ok, I've updated this PR and pushed to the http-message-aware-trait repo, having tested on Ubuntu with PHP 5.6, 7.0, 7.1, 7.2, 7.3, 7.4, 8.0, 8.1, 8.2.

It all works locally and I've also tested using the repository on github.

File: src/Http/Request.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 use Psr\Http\Message\StreamInterface;
 use Psr\Http\Message\UriInterface;
+use mpdf\HttpMessageAwareTrait\HttpRequestTrait;
```
Comment (finwe) [inline src/Http/Request.php:7]: PascalCase, please. Also, order uses alphabetically.

File: composer.json [hunk 2]
```diff
@@ -25,9 +25,10 @@
 		"ext-gd": "*",
 		"ext-mbstring": "*",
 		"mpdf/psr-log-aware-trait": "^2.0 || ^3.0",
+		"mpdf/http-message-aware-trait": "^2.0 || ^3.0",
```
[the remark this answers is not in the document]
Comment (NigelCunningham) [reply composer.json]: Updated.

File: src/Http/Request.php (hunk 1 above)
Comment (NigelCunningham) [reply src/Http/Request.php]: Resolved, thanks.

**Comment by NigelCunningham** [comment]
I believe all issues have been resolved now. I've renamed the repo contents to mpdf/psr-http-message-shim. If you're able to rename the repo as well, everything should just work.

I've been testing with the following addition to the mpdf/mpdf composer.json:

diff --git a/composer.json b/composer.json
index 198b2c9..efc17ed 100644
--- a/composer.json
+++ b/composer.json
@@ -67,5 +67,11 @@
        },
        "config": {
                "sort-packages": true
+       },
+       "repositories": {
+               "mpdf/psr-http-message-shim": {
+                       "type": "vcs",
+                       "url": "https://github.com/mpdf/http-message-aware-trait.git"
+               }
        }
 }
 
 ... and have successfully checked tests pass with PHP 5.6, 7.0 through 7.4 inclusive and 8.0 and 8.1.

**Review by finwe — APPROVED** [review]

**Comment by finwe** [comment]
Great job, thanks!

**Comment by NigelCunningham** [comment]
You're welcome!

**Comment by NigelCunningham** [comment]
Could I just query as to when you're planning another release? I can make the Drupal module that depends on this use the development branch for now, but it would be better to avoid that if I can :)

**Comment by finwe** [comment]
Shortly. I just need to resolve one seemingly minor issue elsewhere to release 8.2.0

**Comment by NigelCunningham** [comment]
Great; thanks!

**Comment by finwe** [comment]
Released.

**Pull request merged**